Mathlib/Order/JordanHolder.lean

```diff
@@ -196,53 +178,20 @@ theorem total {s : CompositionSeries X} {x y : X} (hx : x ∈ s) (hy : y ∈ s)
   exact le_total i j
 #align composition_series.total CompositionSeries.total
 
-/-- The ordered `List X` of elements of a `CompositionSeries X`. -/
-def toList (s : CompositionSeries X) : List X :=
-  List.ofFn s
-#align composition_series.to_list CompositionSeries.toList
-
-/-- Two `CompositionSeries` are equal if they are the same length and
-have the same `i`th element for every `i` -/
-theorem ext_fun {s₁ s₂ : CompositionSeries X} (hl : s₁.length = s₂.length)
-    (h : ∀ i, s₁ i = s₂ (Fin.cast (congr_arg Nat.succ hl) i)) : s₁ = s₂ := by
-  cases s₁; cases s₂
-  -- Porting note: `dsimp at *` doesn't work. Why?
-  dsimp at hl h
-  subst hl
-  simpa [Function.funext_iff] using h
-#align composition_series.ext_fun CompositionSeries.ext_fun
+#align composition_series.to_list RelSeries.toList
+#align composition_series.ext_fun RelSeries.ext
+#align composition_series.length_to_list RelSeries.length_toList
+#align composition_series.to_list_ne_nil RelSeries.toList_ne_empty
 
-@[simp]
-theorem length_toList (s : CompositionSeries X) : s.toList.length = s.length + 1 := by
-  rw [toList, List.length_ofFn]
-#align composition_series.length_to_list CompositionSeries.length_toList
-
-theorem toList_ne_nil (s : CompositionSeries X) : s.toList ≠ [] := by
-  rw [← List.length_pos_iff_ne_nil, length_toList]; exact Nat.succ_pos _
-#align composition_series.to_list_ne_nil CompositionSeries.toList_ne_nil
-
-theorem toList_injective : Function.Injective (@CompositionSeries.toList X _ _) :=
-  fun s₁ s₂ (h : List.ofFn s₁ = List.ofFn s₂) => by
-  have h₁ : s₁.length = s₂.length :=
-    Nat.succ_injective
-      ((List.length_ofFn s₁).symm.trans <| (congr_arg List.length h).trans <| List.length_ofFn s₂)
-  apply ext_fun h₁
-  exact congr_fun <| List.ofFn_injective <| h.trans <|
-    List.ofFn_congr (congr_arg Nat.succ h₁).symm _
+theorem toList_injective : Function.Injective (RelSeries.toList (r := IsMaximal (X := X))) :=
+  fun _ _ h ↦ (RelSeries.Equiv (r := IsMaximal (X := X))).injective <| Subtype.ext h
```

Comment:
It feels a bit odd to have this theorem which doesn't mention `CompositionSeries` here. Can it go upstream?

Reply:
moved, thanks